projects/discussion-ui/src/lib/services/config.service.ts

```diff
@@ -67,6 +68,15 @@ export class ConfigService implements OnInit {
     return this.getContextData
   }
 
+  setCategoryid(id) {
+    this.categoryId = id
+    this.changedSubject.next(id)
```

Comment:
What does this changedSubject do? Maybe we can give a specific name based on what changes it captures